lib/registry.go

```diff
@@ -16,55 +16,72 @@ import (
 
 // RegistryClientMethods defines business logic for working with registries
 type RegistryClientMethods struct {
-	inst *Instance
+	d dispatcher
 }
 
-// NewRegistryClientMethods creates client methods from an instance
-func NewRegistryClientMethods(inst *Instance) *RegistryClientMethods {
-	return &RegistryClientMethods{
-		inst: inst,
-	}
+// Name returns the name of this method group
+func (m RegistryClientMethods) Name() string {
+	return "registry"
 }
 
-// CoreRequestsName implements the Requests interface
-func (RegistryClientMethods) CoreRequestsName() string { return "registry" }
+// Attributes defines attributes for each method
+func (m RegistryClientMethods) Attributes() map[string]AttributeSet {
+	return map[string]AttributeSet{
+		"createprofile":   {denyRPC, ""},
+		"proveprofilekey": {denyRPC, ""},
+	}
+}
 
 // RegistryProfile is a user profile as stored on a registry
 type RegistryProfile = registry.Profile
 
+// RegistryProfileParams encapsulates arguments for creating or proving a registry profile
+type RegistryProfileParams struct {
+	Profile *RegistryProfile
+}
+
 // CreateProfile creates a profile
-func (m RegistryClientMethods) CreateProfile(ctx context.Context, p *RegistryProfile) error {
-	if m.inst.http != nil {
-		return ErrUnsupportedRPC
-	}
+func (m RegistryClientMethods) CreateProfile(ctx context.Context, p *RegistryProfileParams) error {
+	_, _, err := m.d.Dispatch(ctx, dispatchMethodName(m, "createprofile"), p)
+	return dispatchReturnError(nil, err)
+}
+
+// ProveProfileKey sends proof to the registry that this user has control of a
+// specified private key, and modifies the user's config in order to reconcile
+// it with any already existing identity the registry knows about
+func (m RegistryClientMethods) ProveProfileKey(ctx context.Context, p *RegistryProfileParams) error {
+	_, _, err := m.d.Dispatch(ctx, dispatchMethodName(m, "proveprofilekey"), p)
+	return dispatchReturnError(nil, err)
+}
 
+// registryImpl holds the method implementations for RegistryMethods
+type registryImpl struct{}
+
+// CreateProfile creates a profile
+func (registryImpl) CreateProfile(scope scope, p *RegistryProfileParams) error {
 	// TODO(arqu): this should take the profile PK instead of active PK once multi tenancy is supported
-	ownerPk := m.inst.repo.Profiles().Owner().PrivKey
-	pro, err := m.inst.registry.CreateProfile(p, ownerPk)
+	ownerPk := scope.Profiles().Owner().PrivKey
```

Comment:
wasn't sure if this should actually be changed to `scope.ActiveProfile`

Reply:
Per the new setup, scope should be the governing body for the profile in lib and bellow. Think owner is the correct choice by the latest discussions unless we specifically give a sub-profile a "superuser" token. Keeping owner for now makes more sense.